### DISCUSSION TITLE
What is the use case having the down(knex) function in migration scripts?

### DISCUSSION THREAD

**Discussion by superholz** [directus/directus#13927] [Archive]
I wonder what the use case is of having a down function in migration scripts. 

When writing migration script for directus they [have to have an up and a down function](https://docs.directus.io/extensions/migrations/#structure).  So I presume those are mant to be for the case that one would like to revert/rollback a migration. However I'm not aware how one would trigger this rollback for directus.

Background is that using directus me an my colleagues did not really think about it first but where still writing down functions. Depending on the change it can significant effort to write a down(knex) code. If there is no relevant use case we can decide not to writing it.

When a migration fails it is not applied, so I think for that case there is no need for a down function.

**Comment by br41nslug** [comment]
The `down` is indeed meant in case you want to roll back a bad/failed update. You can manually run migrations `up`/`down` using the CLI https://docs.directus.io/reference/cli/#upgrade-the-database

**Reply by superholz** [reply]
Thank you @br41nslug for this quick reply! Do you know how would one use the command to revert a specific migration or chronologically revert everything until a specific migration?

**Reply by br41nslug** [reply]
I don't think you can pick a specific migration. `npx directus database migrate:down` will only revert the latest migration because order matters so you would have to repeat this command until you reach the earliest migration you want to revert. I think this just does one at the time because it would generally be used in error situations where the granular control is useful to revert one migration and test that.
